selftest/uprobe-multi/main.go

```diff
@@ -0,0 +1,229 @@
+package main
+
+import "C"
+import (
+	"bytes"
+	"debug/elf"
+	"encoding/binary"
+	"fmt"
+	"hash/fnv"
+	"log"
+	"os"
+	"strings"
+
+	bpf "github.com/aquasecurity/libbpfgo"
+	"github.com/aquasecurity/libbpfgo/helpers"
+)
+
+type Event struct {
+	Cookie uint64
+}
+
+type FunctionInfo struct {
+	Name   string
+	Offset uint64
+}
+
+const (
+	attachBatchSize  = 256
+	bpfProgramName   = "uprobe__test_functions"
+	bpfProgramObject = "main.bpf.o"
+)
+
+var (
+	excludePatterns = []string{"runtime.", "go:", "internal"}
+)
+
+func main() {
+	if len(os.Args) < 3 {
+		fmt.Fprintln(os.Stderr, "wrong syntax")
+		os.Exit(-1)
+	}
+
+	binaryPath := os.Args[1]
+	expectedSymbolNames := strings.Split(os.Args[2], ",")
+	symbols, err := getFunSyms(binaryPath)
```

Comment:
We actually trace as much functions as possible here, so we test an high number of uprobes, guaranteeing to miss failure by batching, because of the maximum buffer size limited by the related link create BPF syscall's attribute.